backpack-compose/src/main/kotlin/net/skyscanner/backpack/compose/badge/BpkBadge.kt

```diff
@@ -60,26 +63,38 @@ fun BpkBadge(
     Row(
         modifier = modifier
             .semantics(mergeDescendants = true) { }
-            .requiredHeight(BpkSpacing.Lg)
             .border(1.dp, type.borderColor, BadgeShape)
             .background(type.backgroundColor, BadgeShape)
-            .padding(horizontal = BpkSpacing.Md),
+            .padding(horizontal = BpkSpacing.Md, vertical = BpkSpacing.Sm),
         horizontalArrangement = Arrangement.spacedBy(BpkSpacing.Sm),
         verticalAlignment = Alignment.CenterVertically,
     ) {
         val contentColor = type.contentColor
-        if (icon != null) {
+        val badgeIcon = when (type) {
```

Comment:
this will always set the icon to the pre-defined one for the warning, destructive, success types even if the consumer provided a different icon. we should check if the icon is null and only then use the default one instead

Reply:
Oh I think I misunderstood then, I thought the badges were fixed for these types, i.e. always the same icon for Success, Warning and Destructive. I asked Adam if the consumer should be able to change the icon colours and he said no, but maybe I asked the wrong question and the icon should still be optional?

Reply:
they should not be able to change the icon colours, but i believe the icon itself. so we _always_ show an icon for these types, but we only use the default if no other icon is provided

Reply:
Ahh I see! Thanks for clarifying this, I'll fix it now.

Reply:
I think this should be fixed now!